recipes/wasmtime/all/conanfile.py

```diff
@@ -0,0 +1,103 @@
+    description = "Standalone JIT-style runtime for WebAssembly, using Cranelift"
+    topics = ("webassembly", "wasm", "wasi")
+    settings = "os", "compiler", "arch"
+    options = {
+        "shared": [True, False],
+        'fPIC': [True],
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+    }
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "apple-clang": "9.4",
+            "clang": "3.3",
+            "gcc": "4.9.4"
+        }
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        compiler = self.settings.compiler
+        min_version = self._minimum_compilers_version[str(compiler)]
+        try:
+            if tools.Version(compiler.version) < min_version:
+                msg = (
+                    "{} requires C{} features which are not supported by compiler {} {} !!"
+                ).format(self.name, self._minimum_cpp_standard, compiler, compiler.version)
+                raise ConanInvalidConfiguration(msg)
+        except KeyError:
+            msg = (
+                "{} recipe lacks information about the {} compiler, "
+                "support for the required C++{} features is assumed"
+            ).format(self.name, compiler, self._minimum_cpp_standard)
+            self.output.warn(msg)
+
+        if (not (self.version in self.conan_data["sources"]) or
+            not (str(self.settings.os) in self.conan_data["sources"][self.version]) or
+            not (str(self.settings.arch) in self.conan_data["sources"][self.version][str(self.settings.os)] ) ):
+            raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os not available")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version][str(self.settings.os)][str(self.settings.arch)], destination=self._source_subfolder, strip_root=True)
+
+    def package(self):
+        include_path = os.path.join(self._source_subfolder, 'include')
+        self.copy('*.h', dst='include', src=include_path)
+        self.copy('*.hh', dst='include', src=include_path)
+        self.copy('*.hpp', dst='include', src=include_path)
+
+        self.copy('*.lib', dst='lib', keep_path=False)
+        self.copy('*.dll', dst='bin', keep_path=False)
+        self.copy('*.so', dst='lib', keep_path=False)
+        self.copy('*.dylib', dst='lib', keep_path=False)
+        self.copy('*.a', dst='lib', keep_path=False)
```

Comment:
```suggestion
        if self.options.shared:
            if tools.is_apple_os(self.settings.os):
                self.copy("*.dylib", dst="lib", keep_path=False)
            else:
                self.copy("*.so", dst="lib", keep_path=False)
        else:
            self.copy("*.a", dst="lib", keep_path=False)
```

Reply:
I do not know why it is better
The way I've implemented seems much simpler

Reply:
*.dll.lib is not a good exported library name, we already had problems with it. Your project is not the first one using it. Also, separating the copies by shared/static, will avoid mixing wrong extensions according the package.

Reply:
It is already provided in such format ...
I can do nothing with it, at this stage I only contribute recipe for this package
Also I do not like it to be renamed to leave it as is in deployed tar.gz

Reply:
I understand you concern, but you will find the same pattern in other recipes in CCI. Still, it will better to simplify it and keep easy for maintenance.

Reply:
It is not simpler ... It is more complex than needed ...
I have 5 lines, suggested changes are 12 lines